src/main/scala/com/evolutiongaming/util/Validation.scala

```diff
@@ -184,6 +185,28 @@ object Validation {
   implicit class IterableOps[A](val self: Iterable[A]) extends AnyVal {
 
     def ?>>[B](left: => B): Either[B, Iterable[A]] = if (self.isEmpty) Left(left) else Right(self)
+
+    /**
+      * Checks that all elements in `self` are valid by applying the `test` function to each.
+      * If so, returns a [[scala.util.Right]] otherwise returns the first [[scala.util.Left]]
+      * encountered.
+      *
+      * @param test function to apply to each element
+      * @tparam T type of error the `test` function returns
+      */
+    def allValid[T](test: A => Either[T, Unit]): Either[T, Unit] = {
```

Comment:
My idea was to return the first `Left` the `test` function encounters so there is at least some meaningful information of what was wrong. Otherwise we could just do a `_ <- list.forall(...) trueOr "Something was wrong"` and not need this at all.